provisioning/baremetal_pod.go

```diff
@@ -258,28 +260,64 @@ func newMetal3InitContainers(info *ProvisioningInfo) []corev1.Container {
 	}
 
 	initContainers = append(initContainers, createInitContainerMachineOsDownloader(info, true))
+	initContainers = append(initContainers, createInitContainerCoreOSIPA(info))
 
 	return injectProxyAndCA(initContainers, info.Proxy)
 }
 
-func createInitContainerIpaDownloader(images *Images) corev1.Container {
+// getRegistryPullSecret fetches the registry config auth file which is then passed
+// as an env variable to the Ironi-Image's configure-coreos-ipa endpoint. Note that
+// we cannot use LocalObjectReference because the secret is in a different NS.
+func getRegistryPullSecret(info *ProvisioningInfo, targetNamespace string) string {
+	// Check whether the Client is present before using it
+	if info.Client != nil {
+		secretClient := info.Client.CoreV1().Secrets(targetNamespace)
+		secret, err := secretClient.Get(context.TODO(), pullSecretName, metav1.GetOptions{})
```

Comment:
IIUC since 887809db97969ea5eaccacb5713a42a8f13edda3 this will never be in the cache.

Obviously you tried this and it worked. @andfasano does that mean we're always reading directly from the API (i.e. for everything except the Provisioning resource itself) and never the cache?

Reply:
Yes, this Secret in the a Namespace other than "openshift-machine-api" so it will be read directly from the API server and not from the cache.

Reply:
https://github.com/openshift/ironic-image/pull/196 should take care of this above concern I raised.

Reply:
https://bugzilla.redhat.com/show_bug.cgi?id=1986098